qrao/magic_rounding.py

```diff
@@ -99,17 +99,22 @@ class MagicRounding(RoundingScheme):
 
     """
 
-    # This information should really come from the encoding.
-    # Right now it's basically hard coded for one QRAC.
-    _ASSIGN = (
-        {"0": (0, 0, 0), "1": (1, 1, 1)},  # IMI
-        {"0": (0, 1, 1), "1": (1, 0, 0)},  # XMX
-        {"0": (1, 0, 1), "1": (0, 1, 0)},  # YMY
-        {"0": (1, 1, 0), "1": (0, 0, 1)},  # ZMZ
-    )
-
-    # Pauli op string to label index in ops (assumes 3-QRAC)
-    _XYZ = {"X": 0, "Y": 1, "Z": 2}
+    _DECODING = {
+        3: (
+            {"0": [0, 0, 0], "1": [1, 1, 1]},
+            {"0": [0, 1, 1], "1": [1, 0, 0]},
+            {"0": [1, 0, 1], "1": [0, 1, 0]},
+            {"0": [1, 1, 0], "1": [0, 0, 1]},
+        ),
+        2: (
+            {"0": [0, 0], "1": [1, 1]},
+            {"0": [0, 1], "1": [1, 0]},
```

Comment:
I think some comments might be helpful.

```suggestion
        3: (  # Eq. (8)
            {"0": [0, 0, 0], "1": [1, 1, 1]},  # I mu+ I, I mu- I
            {"0": [0, 1, 1], "1": [1, 0, 0]},  # X mu+ X, X mu- X
            {"0": [1, 0, 1], "1": [0, 1, 0]},  # Y mu+ Y, Y mu- Y
            {"0": [1, 1, 0], "1": [0, 0, 1]},  # Z mu+ Z, Z mu- Z
        ),
        2: (  # Sec. VII
            {"0": [0, 0], "1": [1, 1]},  # I xi+ I, I xi- I
            {"0": [0, 1], "1": [1, 0]},  # X xi+ X, X xi- X
```